src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -1468,15 +1468,21 @@ void addSubpatternsForTuple(ImmutableArray<TypeWithAnnotations> elementTypes)
                     memberType = CreateErrorType();
                     member = null;
                     hasErrors = true;
+                    isLengthOrCount = false;
                 }
                 else
                 {
                     member = LookupMembersForPropertyPattern(inputType, expr, diagnostics, ref hasErrors);
                     memberType = member.Type;
+                    isLengthOrCount = memberType.SpecialType == SpecialType.System_Int32 &&
```

Comment:
nit: a comment would be useful here: // if we're dealing with the member that makes the type countable, and the type is also indexable, then it will be assumed to always return a non-negative value